tests/filtered_relation/tests.py

```diff
@@ -279,28 +286,182 @@ def test_as_subquery(self):
         qs = Author.objects.filter(id__in=inner_qs)
         self.assertSequenceEqual(qs, [self.author1])
 
-    def test_with_foreign_key_error(self):
+    def test_with_nested_foreign_key(self):
+        qs = Author.objects.annotate(
+            book_editor_worked_with=FilteredRelation('book__editor', condition=Q(book__title__icontains='book')),
+        ).filter(
+            book_editor_worked_with__isnull=False
+        ).select_related(
+            'book_editor_worked_with'
+        ).order_by(
+            'pk', 'book_editor_worked_with__pk'
+        ).distinct()
+
+        with self.assertNumQueries(1):
+            self.assertQuerysetEqual(qs, [
+                (self.author1, self.editor_a),
+                (self.author2, self.editor_b),
+            ], lambda x: (x, x.book_editor_worked_with))
+
+    def test_with_nested_field(self):
+        qs = Author.objects.annotate(
+            book_editor_worked_with=FilteredRelation('book__editor', condition=Q(book__title__icontains='book')),
+        ).filter(
+            book_editor_worked_with__isnull=False
+        ).values(
+            'name', 'book_editor_worked_with__name'
+        ).order_by(
+            'name', 'book_editor_worked_with__name'
+        ).distinct()
+
+        self.assertSequenceEqual(qs, [
+            {'name': self.author1.name, 'book_editor_worked_with__name': self.editor_a.name},
+            {'name': self.author2.name, 'book_editor_worked_with__name': self.editor_b.name},
+        ])
+
+    def test_with_deep_nested_foreign_key(self):
+        qs = Book.objects.annotate(
+            author_favorite_book_editor=FilteredRelation(
+                'author__favorite_books__editor',
+                condition=Q(author__favorite_books__title__icontains='Jane A')
+            )
+        ).filter(
+            author_favorite_book_editor__isnull=False
+        ).select_related(
+            'author_favorite_book_editor'
+        ).order_by(
+            'pk', 'author_favorite_book_editor__pk'
+        )
+
+        with self.assertNumQueries(1):
+            self.assertQuerysetEqual(qs, [
+                (self.book1, self.editor_b),
+                (self.book4, self.editor_b),
+            ], lambda x: (x, x.author_favorite_book_editor))
+
+    def test_with_foreign_key_on_condition_deeper_than_relationship_error(self):
```

Comment:
@charettes , I've considered your comment and I think that the current limitations around lookup usages prevent that from happening. This test asserts that the "left side" of the condition can't be deeper, which should avoid to N-to-Many relationship nested query issue.